pandas/plotting/_matplotlib/hist.py

```diff
@@ -44,14 +44,25 @@ def _plot(
         stacking_id=None,
         **kwds,
     ):
+
+        y = y[~isna(y)]
+
+        # if bins is a string (e.g. "auto"), get actual bins
+        # so that it can be used in creating stackers below
+        _, bins = np.histogram(
+            y,
+            bins=bins,
+            range=kwds.get("range", None),
+            weights=kwds.get("weights", None),
+        )
+
```

Comment:
Why is this necessary? Why can we just pass the string to matplotlib?

Reply:
The comment above it explains why - we need the value of bins to be numeric because it is used in creating stackers here: https://github.com/pandas-dev/pandas/blob/129b62fbe8e2aa153ecdaccdcc14a8846e430060/pandas/plotting/_matplotlib/hist.py#L59-L62

The results from initializing the stacker are required to create the plot (specifically the argument `bottom`).

Reply:
Matplotlib returns `bins` from `hist`. Can we pass it the string capture that?

Reply:
You're correct that it returns bins (2nd arg here):

https://github.com/pandas-dev/pandas/blob/129b62fbe8e2aa153ecdaccdcc14a8846e430060/pandas/plotting/_matplotlib/hist.py#L65

But that needs `bottom`. `bottom` needs `base` and `base` needs `bins` to be a list not a string (see lines above).